toolkit/pallets/block-rewards-payouts/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet_prelude::*;
+pub use pallet::*;
+use sp_block_rewards_payouts::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_system::pallet_prelude::*;
+	use scale_info::prelude::format;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_block_production_log::Config {
+		/// Type identifying reward receiver on the Partner Chain
+		type BeneficiaryId: Member + Parameter + MaxEncodedLen;
+
+		/// Should return slot up to which rewards should be paid out or None.
+		fn should_pay_out(slot: Slot) -> Option<Slot>;
+	}
+
+	#[pallet::inherent]
+	impl<T: Config> ProvideInherent for Pallet<T> {
+		type Call = Call<T>;
+		type Error = sp_block_rewards_payouts::InherentError;
+		const INHERENT_IDENTIFIER: InherentIdentifier =
+			sp_block_rewards_payouts::INHERENT_IDENTIFIER;
+
+		fn create_inherent(data: &InherentData) -> Option<Self::Call> {
+			let up_to_slot = Self::get_rewards_inherent_data(data)?.up_to_slot;
+
+			Some(Call::note_payout { up_to_slot })
+		}
+
+		fn check_inherent(call: &Self::Call, data: &InherentData) -> Result<(), Self::Error> {
+			let Some(expected_inherent_data) = Self::get_rewards_inherent_data(data) else {
```

Comment:
This code assumes that inherent data is always available. You might want to check and conditionally return an Err?